osg-koji

```diff
@@ -313,11 +313,43 @@ def verify_koji_dir(config_dir):
             (CLIENT_CERT_FILE, "User certificate file (client cert)"),
             (SERVER_CERT_FILE, "Server certificate file (CA bundle)")]:
 
+        fullpath = os.path.join(config_dir, path)
         # Note: use lexists because if we're using grid proxies then the
         # CLIENT_CERT_FILE may be a broken symlink--and that's ok
-        if not os.path.lexists(os.path.join(config_dir, path)):
+        if not os.path.lexists(fullpath):
             raise Error("%s (%s) not found in config directory (%s).\n%s"
                         % (descr, path, config_dir, RUN_SETUP_MSG))
+        # Catch broken client cert symlink separately
+        if path == CLIENT_CERT_FILE:
+            try:
+                target = os.readlink(fullpath)
+                if not os.path.exists(fullpath):
+                    print("%s -> %s is a broken symlink.\n"
+                          "You may need to run grid-proxy-init before doing authenticated operations."
+                          % (fullpath, target),
+                          file=sys.stderr)
+            except OSError as e:
+                if e.errno != 22: raise
+                # 22 = "Invalid argument" -- not a symlink
+
+
+def run_koji(args=None, use_exec=False):
+    """Run koji with the given list of args.  Replaces current process if use_exec is true.
+    Returns return code of os.system() otherwise.
+    Catches missing koji binary.
+    """
+    args = args or []
+    try:
+        if use_exec:
+            os.execlp("koji", "koji", *args)
+        else:
+            cmd = "koji"
+            if args:
+                cmd += " " + " ".join(shell_quote(x) for x in args)
+            return os.system(cmd)
+    except OSError as err:
+        if err.errno == 2:  # file not found
+            raise Error("Couldn't find `koji` binary.  Is koji installed and in your PATH?")
```

Comment:
What happens if there is some other `OSError`?

Reply:
Good call. Fixed.